### PR TITLE
add save index wait duration to index-manager for multiple agent index upload delay

### PR DESCRIPTION
**Pull request by kpango**
Signed-off-by: kpango <kpango@vdaas.org>

### Description:
This PR includes 3 fixes for auto-index functionality of Vald and changed 2 components (index-manager, agent)
1. add save index wait duration configuration to prevent cluster wide frequently upload burst for object storage bandwidth. (index-manager)
2. save index and create index daemon runs separately which enables frequently update index and reduce network bandwidth
3. initialize duration paramas if it's smaller than 0 which converted into 290 years which means never work.
(index-manager, agent)

### Environment:

- Go Version: 1.16.4
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [x] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [x] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/config/index.go [hunk 1]
```diff
@@ -37,6 +37,12 @@ type Indexer struct {
 	// AutoIndexDurationLimit represents auto indexing duration limit
 	AutoIndexDurationLimit string `yaml:"auto_index_duration_limit" json:"auto_index_duration_limit"`
 
+	// AutoSaveIndexDurationLimit represents auto save index duration limit
+	AutoSaveIndexDurationLimit string `yaml:"auto_save_index_duration_limit" json:"auto_save_index_duration_limit"`
+
+	// AutoSaveIndexWaitDuration represents auto save index wait for next duration
+	AutoSaveIndexWaitDuration string `yaml:"auto_save_index_wait_for_next_duration" json:"auto_save_index_wait_for_next_duration"`
```
Comment (rinx) [inline internal/config/index.go:44]: Why is the name different for field and yaml? It is better to use same name.
I think `AutoSaveIndexWaitDuration` is more understandable.

**Review by rinx — CHANGES_REQUESTED** [review]
Please fix the failing tests (for golangci-lint).

File: internal/config/index.go (hunk 1 above)
Comment (kpango) [reply internal/config/index.go]: ok, renamed.

**Comment by kpango** [comment]
please ignore test-code diff.

File: Makefile [hunk 2]
```diff
@@ -369,7 +369,7 @@ format: \
 ## run gofumpt, goimports for all go files
 format/go:
 	find ./ -type d -name .git -prune -o -type f -regex '.*[^\.pb]\.go' -print | xargs gofumpt -w
-	find ./ -type d -name .git -prune -o -type f -regex '.*[^\.pb]\.go' -print | xargs goimports -w
+	find ./ -type d -name .git -prune -o -type f -regex '.*\.go' -print | xargs goimports -w
```
Comment (kevindiu) [inline Makefile:372]: should we ignore formatting generated files?

**Review by kevindiu — COMMENTED** [review]
LGTM after my comment

File: Makefile (hunk 2 above)
Comment (kpango) [reply Makefile:372]: yes, apis/grpc code was already goimports executed code, so if we don't run it will make many difference to this pr.

Comment (kpango) [reply Makefile:372]: and I think it would be okay to run gofumpt for it.

**Review by kevindiu — APPROVED** [review]
LGTM

**Review by hlts2 — APPROVED** [review]

**Review by kevindiu — APPROVED** [review]

**Pull request merged**